ui/job-view/JobView.jsx

```diff
@@ -76,6 +81,49 @@ class JobView extends React.Component {
         ],
       });
     });
+
+    // Only set up the revision polling interval if revision.txt exists on page load
+    this.checkServerRevision().then((revision) => {
+      this.setState({ serverRev: revision });
+      this.updateInterval = setInterval(() => {
+        this.checkServerRevision()
+          .then((revision) => {
+            const { serverChangedTimestamp, serverRev } = this.state;
+            if (this.$rootScope.serverChanged) {
+              if (Date.now() - serverChangedTimestamp > REVISION_POLL_DELAYED_INTERVAL) {
+                this.setState({ serverChangedDelayed: true });
+                // Now that we know there's an update, stop polling.
+                clearInterval(this.updateInterval);
+              }
+            }
+            // This request returns the treeherder git revision running on the server
+            // If this differs from the version chosen during the UI page load, show a warning
+            // Update $rootScope.serverChanged so that the subdued notification
+            // can be shown on the secondary navbar.
+
+            // TODO: Change this to a state/prop var when that secondary bar is
+            // converted to React.
+            if (serverRev && serverRev !== revision) {
+              this.setState({ serverRev: revision });
+              if (this.$rootScope.serverChanged === false) {
+                this.$rootScope.serverChanged = true;
+                this.$rootScope.$apply();
+                this.setState({ serverChangedTimestamp: Date.now() });
+              }
+            }
+          });
+      }, REVISION_POLL_INTERVAL);
+    });
+  }
+
+  checkServerRevision() {
```

Comment:
What do you think about either `fetchServerRevision()` or `fetchDeployedRevision()` ?

Reply:
Sure, the second one sounds great to me.  :)